src/Analyzers/CSharp/Tests/UseParameterNullChecking/UseParameterNullCheckingTests.cs

```diff
@@ -0,0 +1,981 @@
+
+class C
+{
+    void M(string s1, string s2)
+    {
+        if (s1 == null || s2 == null)
+            throw new ArgumentNullException(nameof(s1));
+    }
+}";
+            await new VerifyCS.Test()
+            {
+                TestCode = testCode,
+                FixedCode = testCode,
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestNotOnDefault()
+        {
+            var testCode = @"
+using System;
+
+class C
+{
+    void M(string s)
+    {
+        if ((object)s == default)
+            throw new ArgumentNullException(nameof(s));
+    }
+}";
+            await new VerifyCS.Test()
+            {
+                TestCode = testCode,
+                FixedCode = testCode,
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestWithStringExceptionArgument()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"
+using System;
+
+class C
+{
+    void M(string s)
+    {
+        [|if (s == null)
+            throw new ArgumentNullException(""s"");|]
+    }
+}",
+                FixedCode = @"
+using System;
+
+class C
+{
+    void M(string s!!)
+    {
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestWithStringExceptionArgument2()
+        {
+            await new VerifyCS.Test()
+            {
+                TestCode = @"
+using System;
+
+class C
+{
+    void M(string HelloWorld)
+    {
+        [|if (HelloWorld == null)
+            throw new ArgumentNullException(""Hello"" + ""World"");|]
+    }
+}",
+                FixedCode = @"
+using System;
+
+class C
+{
+    void M(string HelloWorld!!)
+    {
+    }
+}",
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestNotWithUnexpectedExceptionArgument()
+        {
+            var testCode = @"using System;
+
+class C
+{
+    void M(string s)
+    {
+        if (s == null)
+            throw new ArgumentNullException(""banana"");
+    }
+}";
+            await new VerifyCS.Test()
+            {
+                TestCode = testCode,
+                FixedCode = testCode,
+                LanguageVersion = LanguageVersionExtensions.CSharpNext
+            }.RunAsync();
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestNotWithExceptionNoArguments()
+        {
```

Comment:
:(   but i'm ok waiting for feedback.

Reply:
Ah. I forgot about this. I can make this happen.